### PR TITLE
Added use Management API for user data

### PR DESCRIPTION
**Pull request by joshcanhelp**
This PR changes where user information comes from on login. Switching from the Auth API /userinfo to the Management API /users/user_id to get identities. 

* Added WP_Auth0_Api_Client::get_client_token() to quickly get a management token
* Updated WP_Auth0_Api_Client::resend_verification_email() to use this token function
* Replaced WP_Auth0_Api_Client::get_user_info() with WP_Auth0_Api_Client::get_user() to return user information during login (management API access is acquired as part of this upgrade), which fixes identities being returned
* Added read:users as a required scope
* Updated DB upgrade to use new WP_Auth0_Api_Client::get_client_token() function to confirm that a client_credentials token can be returned

### CODE REVIEW

**Comment by joshcanhelp** [comment]
@cocojoe @glena - After our conversation about the Implicit flow use case, I'm wondering if this (and probably a few other things) needs to get re-worked. Implicit users, if they really are blocked from calling the Auth0 server, won't benefit from a client grant, won't be able to use the resend email verification anyways, and won't have the user information that makes it possible to change password, email, etc. Maybe all of that is expected (maybe not?) in which case we probably need to check for that.

File: lib/WP_Auth0_Api_Client.php [hunk 1]
```diff
@@ -204,6 +204,42 @@ public static function get_token( $domain, $client_id, $client_secret, $grantTyp
...
+	 * Get a client_credentials token using default stored connection info
+	 *
+	 * @since 3.4.1
+	 *
+	 * @return bool|string
+	 */
+	public static function get_client_token() {
+
+		$response = wp_remote_post( self::get_endpoint( 'oauth/token' ), array(
+				'headers' => self::get_headers( '', 'application/x-www-form-urlencoded' ),
```
Comment (cocojoe) [inline lib/WP_Auth0_Api_Client.php]: where is `content-type` specified?

Comment (joshcanhelp) [reply lib/WP_Auth0_Api_Client.php]: @cocojoe - `self::get_headers()`

**Comment by joshcanhelp** [comment]
@cocojoe - RE: implicit scope ... it does, look a couple of lines down, you'll see implicit adds additional scopes.

**Comment by joshcanhelp** [comment]
Fixes #348

**Pull request merged**